### ISSUE TITLE
MergeSemantics cannot be undone

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#12323]
The following tests that is removing a `MergeSemantics` widget from the hierarchy is failing. After the widget is removed, the semantics of the child nodes are still merged.

```dart
// Copyright 2017 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';
import 'package:flutter/widgets.dart';
import 'package:flutter_test/flutter_test.dart';

import 'semantics_tester.dart';

void main() {
  testWidgets('MergeSemantics', (WidgetTester tester) async {
    final SemanticsTester semantics = new SemanticsTester(tester);

    // merged
    await tester.pumpWidget(
      new Directionality(
        textDirection: TextDirection.ltr,
        child: new MergeSemantics(
          child: new Row(
            children: <Widget>[
              new Semantics(
                  label: 'test1',
                  textDirection: TextDirection.ltr,
                  child: new Container()
              ),
              new Semantics(
                  label: 'test2',
                  textDirection: TextDirection.ltr,
                  child: new Container()
              )
            ],
          ),
        ),
      ),
    );

    expect(semantics, hasSemantics(new TestSemantics.root(label: 'test1\ntest2')));

    // not merged
    await tester.pumpWidget(
      new Directionality(
        textDirection: TextDirection.ltr,
        child: new Row(
          children: <Widget>[
            new Semantics(
                label: 'test1',
                textDirection: TextDirection.ltr,
                child: new Container()
            ),
            new Semantics(
                label: 'test2',
                textDirection: TextDirection.ltr,
                child: new Container()
            )
          ],
        ),
      ),
    );

    debugDumpSemanticsTree(DebugSemanticsDumpOrder.inverseHitTest);

    // This expect fails, semantics are still merged.
    expect(semantics, hasSemantics(new TestSemantics.root(
      children: <TestSemantics>[
        new TestSemantics.rootChild(id: 1, label: 'test1'),
        new TestSemantics.rootChild(id: 2, label: 'test2'),
      ],
    ), ignoreRect: true, ignoreTransform: true));

    semantics.dispose();
  });
}
```

**Comment by goderbauer** [comment]
Fixed with previous commit.
